### ISSUE TITLE
Queries through the query tab do not work

### ISSUE THREAD

**Issue by jbehrends** [voxpupuli/puppetboard#198]
CentOS 7.1
apache + mod_wsgi
puppetboard v0.1.1 installed via pip
puppetdb v3.1.0

Everything else currently works fine in puppetboard.

When I attempt to run a simply query like:
```
["=", "certname", "test"]
```
 with the "Nodes" button selected and I get the following error:
"The request sent to PuppetDB was invalid. This is usually caused by using an unsupported operator."

I've enabled debugging in the logs, and I see the following from the error_log:
```
[Thu Dec 10 15:46:57.823332 2015] [:error] [pid 27609] DEBUG:pypuppetdb.api:_query called with endpoint: environments, path: None, query: None, limit: None, offset: None, summarize_by None, count_by None, count_filter: None
[Thu Dec 10 15:46:57.823466 2015] [:error] [pid 27609] DEBUG:pypuppetdb.api:_url called with endpoint: environments and path: None
[Thu Dec 10 15:46:57.832075 2015] [:error] [pid 27609] INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): mypuppetdb.host
[Thu Dec 10 15:46:57.836359 2015] [:error] [pid 27609] DEBUG:requests.packages.urllib3.connectionpool:"GET /pdb/query/v4/environments HTTP/1.1" 200 1053
```

**Comment by corey-hammerton** [comment]
Do you have more logs available? It will help with identifying the problem.

**Comment by jbehrends** [comment]
The only logs available are puppetboard's apache access and error logs, and puppetdb's access log.  Is there anything else I'm missing, or that I can provide?

Logs:

```
==> error_log <==
[Thu Dec 10 21:11:50.896140 2015] [:error] [pid 27606] DEBUG:pypuppetdb.api:_query called with endpoint: environments, path: None, query: None, limit: None, offset: None, summarize_by None, count_by None, count_filter: None
[Thu Dec 10 21:11:50.896293 2015] [:error] [pid 27606] DEBUG:pypuppetdb.api:_url called with endpoint: environments and path: None
[Thu Dec 10 21:11:50.904801 2015] [:error] [pid 27606] INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): mypuppetdb.host
[Thu Dec 10 21:11:50.909756 2015] [:error] [pid 27606] DEBUG:requests.packages.urllib3.connectionpool:"GET /pdb/query/v4/environments HTTP/1.1" 200 1053

==> puppetboard_access.log <==
1.2.3.4 - - [10/Dec/2015:21:11:50 -0800] "POST /query HTTP/1.1" 400 5482 "http://mypuppetdb.host/query" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/46.0.2490.86 Safari/537.36"

puppetdb-access.log:
1.2.3.4 - - [10/Dec/2015:21:11:50 -0800] "GET /pdb/query/v4/environments HTTP/1.1" 200 1053 "-" "-"
```

**Comment by corey-hammerton** [comment]
@jbehrends I have been able to reproduce this issue on various environments using Puppetboard. Unfortunately it has proven to be elusive, I suspect that Flask is not properly accepting POST requests for the Query tab endpoint but I cannot be certain since my development environment is working through the coffee dev script.

There have been a number of updates since the 0.1.1 release which may be interfering with troubleshooting. If we cannot fix this by the time we release 0.1.2 it will be my first priority for 0.1.3

**Comment by corey-hammerton** [comment]
Through research and testing I came across http://flask-wtf.readthedocs.org/en/latest/csrf.html and confirmed that on HTTP and mod_wsgi that it thinks the CSRF token is missing or incorrect. 

This is the details that I see:  CSRF token missing or incorrect.

Will continue to investigate but if anybody has ideas or experience with this please come forward
